VideoLocker/src/main/java/org/edx/mobile/http/OauthRefreshTokenAuthenticator.java

```diff
@@ -53,14 +56,30 @@ public OauthRefreshTokenAuthenticator(Context context) {
     public Request authenticate(Route route, final Response response) throws IOException {
         logger.warn(response.toString());
 
-        if (!isTokenExpired(response.peekBody(HttpStatus.OK).string())) {
-            return null;
-        }
-
         final AuthResponse currentAuth = loginPrefs.getCurrentAuth();
         if (null == currentAuth || null == currentAuth.refresh_token) {
             return null;
         }
+        String response_body = response.peekBody(200).string();
+
+        switch (getErrorCode(response_body)) {
+            case TOKEN_EXPIRED_ERROR_MESSAGE:
+                break;
+            case TOKEN_NONEXISTENT_ERROR_MESSAGE:
+            case TOKEN_INVALID_GRANT_ERROR_MESSAGE:
+                // Retry request with the current access_token if the original access_token used in
+                // request does not match the current access_token. This case can occur when
+                // asynchronous calls are made and are attempting to refresh the access_token where
+                // one call succeeds but the other fails. https://github.com/edx/edx-app-android/pull/834
+                if (!response.request().headers().get("Authorization").split(" ")[1].equals(currentAuth.access_token)) {
```

Comment:
Will consider this out of scope for this hotfix for now. I think moving the authenticator to the `isOauthBased` section may create some unwanted side effects.